optimum/neuron/utils/misc.py

```diff
@@ -38,12 +38,21 @@
 from transformers.utils.hub import get_checkpoint_shard_files
 
 from ...utils import logging
+from .import_utils import is_torch_xla_available
 from .require_utils import requires_safetensors
 
 
 logger = logging.get_logger()
 
 
+def should_log() -> bool:
```

Comment:
This is very specific to neuronx_distributed, and not really related to logging. Can you move it under the neuronx_distributed directory, and choose a name that clearly states that this verifies we are in the process of the first worker (that's what I understood).